addon_service/authorized_storage_account/serializers.py

```diff
@@ -83,72 +59,35 @@ def __init__(self, *args, **kwargs):
         "authorized_operations": "addon_service.serializers.AddonOperationSerializer",
     }
 
-    def create(self, validated_data):
+    @staticmethod
+    def get_external_service(
+        external_storage_service: ExternalStorageService, **kwargs
+    ) -> ExternalStorageService:
+        return external_storage_service
+
+    def create_authorized_account(
+        self,
+        external_service: ExternalStorageService,
```

Comment:
I think that if we just make this `external_storage_service`, we can kill `get_external_service` as a whole.

Reply:
We need external_service to set auth credentials in other methods, so I beg to disagree.

Reply:
`process_and_set_auth` has access to the `authorized_account`, which has access to its `external_service` (and to its `credentials_format`)

Reply:
In this case I'll remove `process_and_set_auth` because `create` has become really thin

Reply:
But in this case methods in base class should know what to get:
`external_storage_service`, `external_citations_service`, `external_remote_compute_service` which is unpreferable

Reply:
I'm confused.

The only access to `external_service` I'm seeing in `process_and_set_auth` is to determine the `credentials_format`.

Where would the base class need to know the specific model? Wouldn't the base class just use the `external_service` property that exists for exactly this purpose (or the `credentials_format` property that  similarly references the `external_service` property)?

Reply:
My bad, I have overlooked this

Reply:
No worries, that's what CR is for ;)

Reply:
Fixed, take a look